src/main/java/org/apache/kafka/server/quota/StaticQuotaCallback.java

```diff
@@ -54,13 +80,77 @@ public boolean updateClusterMetadata(Cluster cluster) {
 
     @Override
     public void close() {
-
+        try {
+            storageChecker.stop();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+        }
     }
 
     @Override
     public void configure(Map<String, ?> configs) {
         StaticQuotaConfig config = new StaticQuotaConfig(configs, true);
         quotaMap = config.getQuotaMap();
-        log.info("Configured quota callback with {}", quotaMap);
+        storageQuotaSoft = config.getSoftStorageQuota();
+        storageQuotaHard = config.getHardStorageQuota();
+        storageCheckInterval = config.getStorageCheckInterval();
+        logDirs = config.getLogDirs();
+
+        storageChecker.start();
+        log.info("Configured quota callback with {}. Storage quota (soft, hard): ({}, {}). Storage check interval: {}", quotaMap, storageQuotaSoft, storageQuotaHard, storageCheckInterval);
+    }
+
+    private class StorageChecker implements Runnable {
+        private final Thread storageCheckerThread = new Thread(this, "storage-quota-checker");
+        private volatile boolean running = false;
+
+        void start() {
+            if (!running) {
+                running = true;
+                storageCheckerThread.start();
+            }
+        }
+
+        void stop() throws InterruptedException {
+            running = false;
+            storageCheckerThread.interrupt();
+            storageCheckerThread.join();
+        }
+
+        @Override
+        public void run() {
+            if (StaticQuotaCallback.this.logDirs != null && StaticQuotaCallback.this.storageQuotaSoft > 0 && StaticQuotaCallback.this.storageQuotaHard > 0 && StaticQuotaCallback.this.storageCheckInterval > 0) {
+                while (running) {
+                    try {
+                        long diskUsage = checkDiskUsage();
+                        long previousUsage = StaticQuotaCallback.this.storageUsed.getAndSet(diskUsage);
+                        if (diskUsage != previousUsage) {
+                            StaticQuotaCallback.this.resetQuota.set(true);
+                        }
+                        log.debug("Storage usage checked: {}", StaticQuotaCallback.this.storageUsed.get());
+                        Thread.sleep(TimeUnit.SECONDS.toMillis(StaticQuotaCallback.this.storageCheckInterval));
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        break;
+                    } catch (Exception e) {
+                        log.warn("Exception in storage checker thread", e);
+                    }
+                }
+
+            }
+        }
+
+        private long checkDiskUsage() throws IOException {
+            try {
+                Path folder = Paths.get(logDirs);
```

Comment:
I might be wrong, but it seems like the `log.dirs` property could hold more than one directory name. While I guess it should always be a single one for our cases, it might be something to consider if this plugin is expected to be more generally applicable?

https://kafka.apache.org/26/documentation/#log.dirs

Reply:
Good spot, you're correct, it's a comma separated list.

Reply:
I've raised https://github.com/lulf/kafka-static-quota-plugin/issues/3 @grdryn